### PR TITLE
[GP-Bot] ENG-7336 Fix text campaign showing 'Sent' before being sent

### PR DESCRIPTION
**Pull request by Collin-GP**
## Summary
- Fixes a bug where text campaigns displayed "Sent" status even when scheduled for a future time
- The issue occurred because the `p2pJob.status === 'active'` condition was incorrectly interpreted as "sent"
- Now compares the scheduled date/time with the current time before showing "Sent"

## Root Cause
In `OutreachTable.tsx`, the status display logic showed "Sent" (mapped from `completed`) whenever the P2P job status was `active`. However, `active` means the job is ready to run or currently running - not that messages have been delivered. A job can be `active` but scheduled for a future time.

## Fix
Added a `isScheduledDatePassed()` helper function that compares the scheduled date with the current time:
- If the P2P job is active AND the scheduled date has passed → show "Sent"
- If the P2P job is active AND the scheduled date is in the future → show "Scheduled"
- For non-active job statuses, use the outreach status as before

## Test plan
- [x] Added comprehensive unit tests for the status display logic
- [x] Tests cover: active job with past date (Sent), active job with future date (Scheduled), various outreach statuses
- [x] All 234 tests in the codebase pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Small, localized change to status-label rendering logic with comprehensive unit tests; no auth, persistence, or backend behavior changes.
> 
> **Overview**
> Fixes a UI bug in `OutreachTable` where P2P campaigns with `p2pJob.status === "active"` were always shown as **Sent**, even if scheduled in the future.
> 
> Adds `isScheduledDatePassed()` and updates `getP2pStatusLabel()` to show **Scheduled** until the outreach `date` is reached (then **Sent**), while preserving the existing status mapping for non-active jobs. Adds a new `OutreachTable.test.tsx` suite covering past/future/exact/nullable dates, non-active statuses, and `n/a` edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ffa73afd67618f9b0e5017d0e6533f7eb1ee6b42. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by tomer-tgp** [comment]
Task linked: [ENG-7336 Text campaign showed 'sent' before being sent](https://app.clickup.com/t/90132012119/ENG-7336)

**Comment by feliksalbright** [comment]
@cursor review

File: app/dashboard/outreach/components/OutreachTable.tsx [hunk 1]
```diff
@@ -56,24 +56,34 @@ const isStatusKey = (key: string | null | undefined): key is StatusKey => {
...
     return null
   }
 
-  // If P2P job is active, show as completed; otherwise use the outreach status
-  const displayStatus: StatusKey =
-    p2pJob.status === 'active' ? 'completed' : status
-  return statusLabels[displayStatus]
+  // If P2P job is active, only show as completed if the scheduled date has passed
+  // Otherwise show as scheduled (the job is active but hasn't started sending yet)
+  if (p2pJob.status === 'active') {
```
Comment (tomer-tgp) [inline app/dashboard/outreach/components/OutreachTable.tsx:81]: Nit: Can we use a constant instead of a magic string?
